src/ocean/core/Option.d

```diff
@@ -0,0 +1,236 @@
+/******************************************************************************
+
+    Wraps any type in a struct that also contains boolean field indicating if
+    value is in defined state.
+
+    If T is a value type, `Option!(T)` is value type too.
+
+    Copyright:
+        Copyright (c) 2009-2016 Sociomantic Labs GmbH.
+        All rights reserved.
+
+    License:
+        Boost Software License Version 1.0. See LICENSE_BOOST.txt for details.
+        Alternatively, this file may be distributed under the terms of the Tango
+        3-Clause BSD License (see LICENSE_BSD.txt for details).
+
+
+*******************************************************************************/
+
+module ocean.core.Option;
+
+version(UnitTest)
+{
+    import ocean.core.Test;
+}
+
+/// ditto
+struct Option ( T )
```

Comment:
Is there some overload for `cast(bool)` conversion? Maybe it should default to `defined`? Or be disabled, and force users to use `isDefined`?

Reply:
Talking with @david-eckardt-sociomantic seems that there's no implicit cast to bool for the struct anyway, and that `opCast` covers _explicit_ casts, so this is not very useful to have.